### ISSUE TITLE
Update GitLab integration with 0.16.0

### ISSUE THREAD

**Issue by kzantow** [anchore/grype#381]
After: #334 

Once we implement this, we have an opportunity to **update Grype within our GitLab Container Scanning integration**. You'll see in the [Container Scanning docs](https://docs.gitlab.com/ee/user/application_security/container_scanning/#customizing-the-container-scanning-settings) show that the setting `CS_REGISTRY_INSECURE` is supported by only Trivy, not Grype. So once Grype has this configuration option, we can open an MR with GitLab to support the related GitLab configuration option and propose a GitLab feature documentation update.

(The integration code itself lives here: https://gitlab.com/gitlab-org/security-products/analyzers/container-scanning)

**Comment by luhring** [comment]
Specific steps for this:

1. Fork [container-scanning](https://gitlab.com/gitlab-org/security-products/analyzers/container-scanning). And ensure that your fork is configured to automatically receive the new commits from the source repo's `master` branch.
2. Follow the documentation for ["Scanner Updates"](https://gitlab.com/gitlab-org/security-products/analyzers/container-scanning/-/tree/master#scanner-updates) (GitLab glossary: Grype is considered a "scanner", and so is Trivy.) (**Side note**: it looks like GitLab has recently [updated this process](https://gitlab.com/gitlab-org/security-products/analyzers/container-scanning/-/commit/57dbd55d3e14219e6ca50348f242c408cc927751). It looks nice! We should at least consider something similar as we evolve our integration/tool update strategy.)
3. Update the [Grype-wrapping logic](https://gitlab.com/gitlab-org/security-products/analyzers/container-scanning/-/blob/master/lib/gcs/grype.rb) to leverage this new option in Grype and map it back to the GitLab setting `CS_REGISTRY_INSECURE`. Take a look at the [`environment` method](https://gitlab.com/gitlab-org/security-products/analyzers/container-scanning/-/blob/master/lib/gcs/grype.rb#L30-41) to see how we've been handling this mapping of the GitLab environment variables that are prepared in [this class](https://gitlab.com/gitlab-org/security-products/analyzers/container-scanning/-/blob/master/lib/gcs/environment.rb) to Grype-specific config env vars.
4. [Open a Merge Request](https://gitlab.com/danluhring/container-scanning/-/merge_requests/new). **In your MR description**: Mention that this update means the [Container Scanning docs for settings variables](https://docs.gitlab.com/ee/user/application_security/container_scanning/#customizing-the-container-scanning-settings) should be updated, specifically that the text "Trivy. The registry must listen on port 80/tcp." is no longer needed and can be changed to "All". Volunteer to make the docs update yourself (and ask about the latest process for doing so), but there's a chance they'll handle it on their end.

Let me know if you have any questions! The GitLab team in charge of Container Scanning is a great group of people that are easy to work with and keep things moving forward.

**Comment by luhring** [comment]
Grype [v0.16.0](https://github.com/anchore/grype/releases/tag/v0.16.0) was just released today 🚀

**Comment by kzantow** [comment]
There's a MR open with this change now: https://gitlab.com/gitlab-org/security-products/analyzers/container-scanning/-/merge_requests/2579
